StRoot/StETofCalibMaker/StETofCalibMaker.cxx

```diff
@@ -2609,3 +2667,267 @@ StETofCalibMaker::writeHistograms()
+
+   bool fileZero = false;
+
+   //Clean up last entry first
+   for(int i =0; i< 1728;i++){
+     mStateVec[i].clear();
+     mStartVec[i].clear();
+     mGet4StateMap[i] = 0;
+   }
+   mStateMapStart=0;
+   mStateMapStop=0;
+   mDbEntryStop=0;
+   mMasterStartVec.clear();
+   mMasterStartVec.resize(0);
+
+   std::vector< unsigned long int > intVec;
+   
+   //first read
+    if(forward == 0) mGlobalCounter = 1;
+    //jump forward
+    else if(forward > 0) mGlobalCounter++;
+    //jump backward
+    else mGlobalCounter--; // forward < 0
+    
+    if(mGlobalCounter == 0){
+      mGlobalCounter++;
+      fileZero = true;
+    }
+    
+    if(mFileNameGet4State.empty()){
+         
+      TDataSet* dbDataSet = nullptr;
+      dbDataSet = GetDataBase( "Calibrations/etof/etofGet4StateMap" );
+      const int intsPerEntry = 1000000;
+      
+      St_etofGet4StateMap* etofStateMap = static_cast< St_etofGet4StateMap* > ( dbDataSet->Find( "etofGet4StateMap" ) );
+      if( !etofStateMap ) {
+	LOG_ERROR << "unable to get the get4 state map from the database" << endm;	  
+	return;
+      }
+      
+      etofGet4StateMap_st* stateMapTable = etofStateMap->GetTable();
+      
+      for( size_t i=0; i< intsPerEntry; i++ ) {
+	if(stateMapTable->etofGet4State[ i ] <= 0) break; 
+	intVec.push_back(  stateMapTable->etofGet4State[ i ]);
+      } 
+          
+    }else{
+           
+      std::ifstream paramFile;
+      
+      paramFile.open( mFileNameGet4State.c_str() );
+      
+      if( !paramFile.is_open() ) {
+	LOG_ERROR << "unable to get the 'Get4State' parameters from file --> file does not exist" << endm;
+       	return;
+      }
+      
+      unsigned long int temp;
+      while( paramFile >> temp ) {
+	intVec.push_back( temp );
+      }        
+    }
+    
+    std::vector<unsigned long int> startVec;
+    std::map<unsigned long int,vector<int>> stateVec;
+    std::map<unsigned long int ,vector<int>> get4IdVec;
+    
+    unsigned long int lastEvtId =0;
+    
+    for(unsigned int i = 0; i < intVec.size(); i++){
+      
+      // decode nonZero/stateChange ints ( int = 42.xxx.xxx.xxx = 2 states only)
+	switch (intVec.at(i) / 100000000) {
+		
+	case 42	:	
+	int tmp       = intVec.at(i) % 4200000000;
+	int stateInt1 = tmp / 10000;
+	int stateInt2 = tmp % 10000;
+	
+	int Get4Id1 = -1;
+	int get4state1 = -1;
+	int Get4Id2 = -1;
+	int get4state2 = -1;
+	
+	if(stateInt1 < 6912){
+	  Get4Id1 = statInt1 % 1728;
+	  get4state1 = stateInt1 / 1728;
+	}
+	if(stateInt2 < 6912){
+	  Get4Id2 = statInt2 % 1728;
+	  get4state2 = stateInt2 / 1728;
+	}
+	
+	if(i < 864){
+	  
+	  mGet4StateMap[Get4Id1] = get4state1;
+	  mGet4StateMap[Get4Id2] = get4state2;
+	  mGet4ZeroStateMap[Get4Id1] = get4state1;
+	  mGet4ZeroStateMap[Get4Id2] = get4state2;
+	}
+	stateVec[lastEvtId].push_back(get4state1);
+	get4IdVec[lastEvtId].push_back(Get4Id1);
+	stateVec[lastEvtId].push_back(get4state2);
+	get4IdVec[lastEvtId].push_back(Get4Id2);
```

Comment:
`readGet4State` function is too long. Please move the code decoding `intVec` elements into a separate function.

Reply:
moved decoding to separate function